pkg/k8s/watchers/watcher.go

```diff
@@ -372,17 +372,19 @@ func (k *K8sWatcher) resourceGroups() []string {
 	k8sGroups = append(k8sGroups, K8sAPIGroupEndpointV1Core)
 
 	resourceToGroupMapping := map[string]string{
-		synced.CRDResourceName(v2.CNPName):        k8sAPIGroupCiliumNetworkPolicyV2,
-		synced.CRDResourceName(v2.CCNPName):       k8sAPIGroupCiliumClusterwideNetworkPolicyV2,
-		synced.CRDResourceName(v2.CEPName):        k8sAPIGroupCiliumEndpointV2, // ipcache
-		synced.CRDResourceName(v2.CNName):         k8sAPIGroupCiliumNodeV2,
-		synced.CRDResourceName(v2.CIDName):        "SKIP", // Handled in pkg/k8s/identitybackend/
-		synced.CRDResourceName(v2.CLRPName):       k8sAPIGroupCiliumLocalRedirectPolicyV2,
-		synced.CRDResourceName(v2.CEWName):        "SKIP", // Handled in clustermesh-apiserver/
-		synced.CRDResourceName(v2alpha1.CENPName): k8sAPIGroupCiliumEgressNATPolicyV2,
-		synced.CRDResourceName(v2alpha1.CESName):  k8sAPIGroupCiliumEndpointSliceV2Alpha1,
-		synced.CRDResourceName(v2alpha1.CCECName): k8sAPIGroupCiliumClusterwideEnvoyConfigV2Alpha1,
-		synced.CRDResourceName(v2alpha1.CECName):  k8sAPIGroupCiliumEnvoyConfigV2Alpha1,
+		synced.CRDResourceName(v2.CNPName):           k8sAPIGroupCiliumNetworkPolicyV2,
+		synced.CRDResourceName(v2.CCNPName):          k8sAPIGroupCiliumClusterwideNetworkPolicyV2,
+		synced.CRDResourceName(v2.CEPName):           k8sAPIGroupCiliumEndpointV2, // ipcache
+		synced.CRDResourceName(v2.CNName):            k8sAPIGroupCiliumNodeV2,
+		synced.CRDResourceName(v2.CIDName):           "SKIP", // Handled in pkg/k8s/identitybackend/
+		synced.CRDResourceName(v2.CLRPName):          k8sAPIGroupCiliumLocalRedirectPolicyV2,
+		synced.CRDResourceName(v2.CEWName):           "SKIP", // Handled in clustermesh-apiserver/
+		synced.CRDResourceName(v2alpha1.CENPName):    k8sAPIGroupCiliumEgressNATPolicyV2,
+		synced.CRDResourceName(v2alpha1.CESName):     k8sAPIGroupCiliumEndpointSliceV2Alpha1,
+		synced.CRDResourceName(v2alpha1.CCECName):    k8sAPIGroupCiliumClusterwideEnvoyConfigV2Alpha1,
+		synced.CRDResourceName(v2alpha1.CECName):     k8sAPIGroupCiliumEnvoyConfigV2Alpha1,
+		synced.CRDResourceName(v2alpha1.BGPPName):    "SKIP", // Handled in BGP control plane
+		synced.CRDResourceName(v2alpha1.BGPPoolName): "SKIP", // Handled in BGP control plane
```

Comment:
Does this mean that the k8s watcher initialization is done separately from this code? I would much prefer if we can avoid skipping here if possible, since this is the mechanism that we use to properly disable k8s resource sync unconditionally across the codebase, so any exceptions can end up having repercussions on builds where we explicitly intend to disable certain functionality (for instance, in EKSA we use this mechanism to remove features that downstream do not wish to support, which includes this BGP support for now).

Feel free to hit me up for more context on this, maybe that'll help to figure out a good way to make this more maintainable.

Reply:
Might be worth cross-linking the remarks to https://github.com/cilium/cilium/issues/19376 

(Adding this comment so that at least it will show up in the Github UI for that ticket :) )

Reply:
@joestringer we skip the watchers registration here due to the BGP control plane managing its own informers and listers: 
https://github.com/cilium/cilium/blob/636edd09da749f5fbeea2ae7c5eaa2e97d4d682e/pkg/bgpv1/agent/controller.go#L150-L172

This was an initial attempt to avoid head of line blocking in the watcher code, and an implementation of: https://docs.google.com/document/d/1gybVpFOrnjJFLZEMwp_VQJZGQSZ30_pVo2VA2fylBFM/edit#

Reply:
:+1: okay cool, maybe I'll just hit you up then to see if there's a way to control the k8s registration all from one place then with this model then, because otherwise it'll be a pain to continue to support things like EKS-A going forwards if we have to track each feature individually registering its k8s watchers all over the codebase. Background is this PR: https://github.com/cilium/cilium/pull/17677

Reply:
^ We discussed out of band, it's on me to come up with a concrete proposal of how we might improve this.